plugins/modules/ec2_instance.py

```diff
@@ -1699,6 +1751,48 @@ def handle_existing(existing_matches, state):
     return result
 
 
+def enforce_count(existing_matches, module):
+    exact_count = module.params.get('exact_count')
+
+    try:
+        current_count = len(existing_matches)
+        if current_count == exact_count:
+            module.exit_json(
+                changed=False,
+                msg='{0} instances already running, nothing to do.'.format(exact_count)
+            )
+
+        elif current_count < exact_count:
+            to_launch = exact_count - current_count
+            module.params['to_launch'] = to_launch
+            # launch instances
+            try:
+                ensure_present(existing_matches=existing_matches, desired_module_state='present')
+            except botocore.exceptions.ClientError as e:
+                module.fail_json(e, msg='Unable to launch instances')
+        elif current_count > exact_count:
+            to_terminate = current_count - exact_count
+            # get the instance ids of instances with the count tag on them
+            all_instance_ids = sorted([x['InstanceId'] for x in existing_matches])
```

Comment:
How about sorting based on `LaunchTime` instead?  That would mean that the oldest instances go away first, rather than it being (from a customer perspective) near random with some instances almost never going away.

Reply:
This sounds good, having random termination is not really helpful for users I believe.